cache_test.go

```diff
@@ -0,0 +1,144 @@
+package cedar
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/utility"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEnvironmentCache(t *testing.T) {
+	t.Run("PutNewGetAndDelete", func(t *testing.T) {
+		cache := newEnvironmentCache()
+
+		require.True(t, cache.PutNew("key0", "val0"))
+		require.True(t, cache.PutNew("key1", 1))
+
+		val0, ok := cache.Get("key0")
+		assert.True(t, ok)
+		assert.Equal(t, "val0", val0)
+
+		val1, ok := cache.Get("key1")
+		assert.True(t, ok)
+		assert.Equal(t, 1, val1)
+
+		cache.Delete("key0")
+		val0, ok = cache.Get("key0")
+		assert.False(t, ok)
+		assert.Nil(t, val0)
+	})
+	t.Run("WithUpdater", func(t *testing.T) {
```

Comment:
Should this also test other error cases like the case of attempting to register an updater without an initial value + the case of registering multiple updaters?